src/components/base/AccountsList/AccountRow.js

```diff
@@ -11,108 +11,91 @@ import Radio from 'components/base/Radio'
 import CryptoCurrencyIcon from 'components/CryptoCurrencyIcon'
 import FormattedVal from 'components/base/FormattedVal'
 import Input from 'components/base/Input'
-import IconEdit from 'icons/Edit'
-import IconCheck from 'icons/Check'
-import type { T } from 'types/common'
+import { MAX_ACCOUNT_NAME_SIZE } from 'config/constants'
 
 type Props = {
   account: Account,
   isChecked: boolean,
   isDisabled?: boolean,
-  onClick: Account => void,
-  onAccountUpdate: Account => void,
-  t: T,
+  autoFocusInput?: boolean,
+  accountName: string,
+  onToggleAccount?: (Account, boolean) => void,
+  onEditName?: (Account, string) => void,
 }
 
-type State = {
-  isEditing: boolean,
-  accountNameCopy: string,
-}
-
-export default class AccountRow extends PureComponent<Props, State> {
-  state = {
-    isEditing: false,
-    accountNameCopy: '',
-  }
-
-  componentDidUpdate(prevProps: Props, prevState: State) {
-    const startedEditing = !prevState.isEditing && this.state.isEditing
-    if (startedEditing) {
-      this._input && this._input.handleSelectEverything()
-    }
+export default class AccountRow extends PureComponent<Props> {
+  handlePreventSubmit = (e: SyntheticEvent<*>) => {
+    e.preventDefault()
+    e.stopPropagation()
   }
 
-  handleEditClick = (e: SyntheticEvent<any>) => {
-    this.handlePreventSubmit(e)
-    const { account } = this.props
-    this.setState({ isEditing: true, accountNameCopy: account.name })
+  onToggleAccount = () => {
+    const { onToggleAccount, account, isChecked } = this.props
+    if (onToggleAccount) onToggleAccount(account, !isChecked)
   }
 
-  handleSubmitName = (e: SyntheticEvent<any>) => {
-    this.handlePreventSubmit(e)
-    const { account, onAccountUpdate, isChecked, onClick } = this.props
-    const { accountNameCopy } = this.state
-    const updatedAccount = { ...account, name: accountNameCopy }
-    this.setState({ isEditing: false, accountNameCopy: '' })
-    onAccountUpdate(updatedAccount)
-    if (!isChecked) {
-      onClick(updatedAccount)
-    }
+  handleChangeName = (name: string) => {
+    const { onEditName, account } = this.props
+    if (onEditName) onEditName(account, name)
   }
 
-  handlePreventSubmit = (e: SyntheticEvent<any>) => {
-    // prevent account row to be submitted
+  onClickInput = (e: SyntheticEvent<*>) => {
```

Comment:
we can have unified `handleStopEvent` which will be used in place of `handlePreventSubmit` + `onClickInput` (nvm, not so grave)